### PR TITLE
vm: single pass encryption support

### PR DESCRIPTION
**Pull request by yugangw-msft**
Old 2 passes flow is still supported through detecting mechanism proposed by ADE team. Supported linux disto list is also updated.

//CC @SudhakaraReddyEvuri 

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/6186
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py [hunk 1]
```diff
@@ -21,32 +22,49 @@
...
+        r = next((e for e in exts if e.type and e.type.lower().startswith(ade_ext_info['publisher'].lower()) and
+                  e.name.lower() == ade_ext_info['name'].lower()), None)
+    else:
+        exts = vm.resources or []
+        r = next((e for e in exts if (e.publisher.lower() == ade_ext_info['publisher'].lower() and
+                                      e.virtual_machine_extension_type.lower() == ade_ext_info['name'].lower())), None)
+    return r
+
+
+# return (has_new_ade, has_old_ade)
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: Remove?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py [hunk 2]
```diff
@@ -102,30 +125,40 @@ def encrypt_vm(cmd, resource_group_name, vm_name,  # pylint: disable=too-many-lo
...
         'EncryptionOperation': 'EnableEncryption' if not encrypt_format_all else 'EnableEncryptionFormatAll',
         'KeyEncryptionKeyURL': key_encryption_key,
         'KeyEncryptionAlgorithm': key_encryption_algorithm,
         'SequenceVersion': sequence_version,
     }
-    private_config = {
+    if use_new_ade:
+        public_config.update({
+            "KeyVaultResourceId": disk_encryption_keyvault,
+            "KekVaultResourceId": key_encryption_keyvault if key_encryption_key else '',
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: Typo 'KekVaultResourceId' I assume

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py [hunk 3]
```diff
@@ -136,75 +169,71 @@ def encrypt_vm(cmd, resource_group_name, vm_name,  # pylint: disable=too-many-lo
...
-    if not (extension_result.instance_view.statuses and
-            extension_result.instance_view.statuses[0].message):
-        raise CLIError('Could not found url pointing to the secret for disk encryption')
 
-    # 3. update VM's storage profile with the secrets
-    status_url = extension_result.instance_view.statuses[0].message
+    if not use_new_ade:
+        if not (extension_result.instance_view.statuses and
+                extension_result.instance_view.statuses[0].message):
+            raise CLIError('Could not found url pointing to the secret for disk encryption')
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: Typo: 'Could not find'

Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: Thanks for catching this for me

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py (hunk 2 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: This time, it is a not a typo. `kek` stands for `key-vault-key` for securer encryption

**Comment by yugangw-msft** [comment]
@SudhakaraReddyEvuri, can you please post updates on windows 2016 bitlock issue we have faced before we merge the PR. If it stays not fixed, we need to call out through some means as the ratio of hitting this issue is way too high

**Pull request merged**